applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClientBuilder.java

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.applicationconfig;
+
+import com.azure.common.http.HttpClient;
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.policy.AsyncCredentialsPolicy;
+import com.azure.common.http.policy.HttpLogDetailLevel;
+import com.azure.common.http.policy.HttpLoggingPolicy;
+import com.azure.common.http.policy.HttpPipelinePolicy;
+import com.azure.common.http.policy.RetryPolicy;
+import com.azure.common.http.policy.UserAgentPolicy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Provides configuration options for instances of {@link ConfigurationClient}.
+ *
+ * @see ConfigurationClient
+ * @see ConfigurationClientCredentials
+ */
+public final class ConfigurationClientBuilder {
+    //TODO: Fix with https://github.com/Azure/azure-sdk-for-java/issues/3141
+    static final String SDK_NAME = "Azure-Application-Configuration";
+    static final String SDK_VERSION = "1.0.0-SNAPSHOT";
```

Comment:
It might be nice to extract this out into a package-private class, because we will want to centralise this for the sync client too. I would like to have a standard location for this for all client libraries, e.g. some `AzureConfig` or whatever class in the root package.